install_sct

```diff
@@ -637,9 +637,8 @@ fi
 (
   # We use "--ignore-installed" to preserve the version of `certifi` installed into the conda
   # env, which prevents https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3609
-  pip install -r "$REQUIREMENTS_FILE" --ignore-installed certifi &&
```

Comment:
Keep in mind that SCT is currently using 2 different types of requirements file:

* `requirements.txt`: This is the default one, present on `master`, with flexible requirements.
* `requirements_freeze.txt`: This is a set of frozen dependencies for stable releases. It is generated at time of release to ensure a stable set of requirements.

You can see the logic to parse `requirements_freeze.txt` files here:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/6ca81d72c9e3eef265ba3a0ff63d8297b4372f3d/install_sct#L629-L636

Because of this, the changes in this PR will currently break any future releases we create. 

But, I think this is fixable! I think we could put the `requirements_freeze.txt` logic inside of `setup.py`. Something like "if requirements_freeze.txt is present, read the dependencies from file, otherwise use the default requirements".

Reply:
Gotcha, I will submit the logic to parse the frozen file inside setup.py tonight. May I ask, how is the frozen file generated? Do you create a venv, install packages and pip freeze?

Reply:
@joshuacwnewton if the build passes (at least It Works On My Machine™), it's ready for review.

I added support for frozen requirements and improved dependency check.

Reply:
Thank you! I'll take a look at this first thing tomorrow. :)